src/m365/spo/commands/site/site-appcatalog-list.spec.ts

```diff
@@ -0,0 +1,127 @@
+      "ErrorMessage": "Success",
+      "SiteID": "2f9fd04d-2674-40ca-9ad8-d7f982dce5d0"
+    }
+  ];
+
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+  let loggerLogToStderrSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    auth.service.spoUrl = 'https://contoso.sharepoint.com';
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+    auth.service.spoUrl = undefined;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.SITE_APPCATALOG_LIST), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['AbsoluteUrl', 'SiteID']);
+  });
+
+  it('retrieves site collection app catalogs within the tenant (debug)', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === 'https://contoso.sharepoint.com/_api/Web/TenantAppCatalog/SiteCollectionAppCatalogsSites') {
+        return { value: appCatalogResponseValue };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: {} });
+    assert(loggerLogSpy.calledWith(appCatalogResponseValue));
+  });
+
+  it('correctly handles no site collection app catalogs in the tenant', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === 'https://contoso.sharepoint.com/_api/Web/TenantAppCatalog/SiteCollectionAppCatalogsSites') {
+        return { value: [] };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { verbose: true } });
+    assert(loggerLogToStderrSpy.calledWith('No site collection app catalogs found'));
+  });
+
+  it('correctly handles error when retrieving site collection app catalogs', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === 'https://contoso.sharepoint.com/_api/Web/TenantAppCatalog/SiteCollectionAppCatalogsSites') {
+        throw 'Invalid request';
```

Comment:
Let's use a realistic error, for this API, I receive an error in following format:

```suggestion
        throw { error: { error: { message: 'Something went wrong' } }};
```